modules/provision/debug.go

```diff
@@ -0,0 +1,112 @@
+package provision
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zosv2/modules/stubs"
+	"github.com/threefoldtech/zosv2/modules/zinit"
+
+	"github.com/pkg/errors"
+)
+
+// Debug provision schema
+type Debug struct {
+	Host    string `json:"host"`
+	Port    int    `json:"port"`
+	Channel string `json:"channel"`
+}
+
+func debugProvision(ctx context.Context, reservation *Reservation) (interface{}, error) {
+	var cfg Debug
+	if err := json.Unmarshal(reservation.Data, &cfg); err != nil {
+		return nil, err
+	}
+
+	return startZLF(ctx, reservation.ID, cfg)
+}
+
+func debugDecommission(ctx context.Context, reservation *Reservation) error {
+	return stopZLF(ctx, reservation.ID)
+}
+
+func startZLF(ctx context.Context, ID string, cfg Debug) (string, error) {
+	zbus := GetZBus(ctx)
+	identity := stubs.NewIdentityManagerStub(zbus)
+
+	path, err := exec.LookPath("zlf")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to start zlf")
+	}
+
+	z := zinit.New("")
+	if err := z.Connect(); err != nil {
+		return "", errors.Wrap(err, "fail to connect to zinit")
+	}
+	defer z.Close()
+
+	channel := fmt.Sprintf("%s-logs", identity.NodeID().Identity())
+	if cfg.Channel != "" {
+		channel = cfg.Channel
+	}
+
+	s := zinit.InitService{
+		Exec:    fmt.Sprintf("%s --host %s --port %d --channel %s", path, cfg.Host, cfg.Port, channel),
+		Oneshot: false,
+		After:   []string{"networkd"},
```

Comment:
You also need to set `log: stdout` otherwise the logs of the zlf will go to the ringbuffer, which will cause a positive feedback loop

Reply:
I can see that `zlf` only print max 2 lines of log messages on start, so it will not cause a positive feedback loop. this can happen though if zlf start logging on each 'forward' operation.

Reply:
better safe then sorry